sqlglot/parser.py

```diff
@@ -2482,23 +2488,28 @@ def _parse_replace(self):
         self._match_r_paren()
         return columns
 
-    def _parse_csv(self, parse):
-        parse_result = parse()
+    def _parse_csv(self, parse_method):
+        parse_result = parse_method()
         items = [parse_result] if parse_result is not None else []
 
         while self._match(TokenType.COMMA):
-            parse_result = parse()
+            self._attach_comment(parse_result, self._prev.comment)
+            parse_result = parse_method()
             if parse_result is not None:
                 items.append(parse_result)
 
+        # This makes sure we don't attach a comment twice to the last term in a binary expression
+        if not isinstance(parse_result, exp.Binary):
```

Comment:
why would this happen?

Reply:
```sql
SELECT a /*x*/ + b /*x*/
```
This would otherwise wrongly produce:
```sql
SELECT a /* x */ + b /* x */ /* x */
```
That's because we're adding the `/* x */` comment on the Binary node which is printed in the generator's `expressions` method, and we also print it for the primary `b` value. Let me know if you have any ideas on how we can make this cleaner.

Reply:
sorry i don't understand why that would happen

in select a / * x */ + b /* y*/

the comments would be directly attached to a and b, there should be no issues

Reply:
Right, sorry, maybe I miscommunicated that. I'll try to explain it a little better:

- We want to preserve comments for SELECT's expressions, which is currently handled in `Generator.expressions`.

- If we want to be able to preserve comments for the children of a Binary expression, we also have to add comment generation logic in the corresponding methods of the Generator (here: `Generator.binary`). This is what I did to have `SELECT a /*x*/ + b /*y*/` transpile into itself.

- However, this last change is conflicting with the logic for attaching comments in SELECT's expressions. To see this, let's suppose that we didn't have the instance check for Binary in the `_parse_csv` method. Then, `/*y*/` would be attached to both the `Binary` node _and_ the `Identifier` node for `b`, which means that we would generate it twice:

```python
>>> import sqlglot
>>> sqlglot.transpile("SELECT a /*x*/ + b /*y*/")[0] # without the isinstance check
'SELECT a /*x*/ + b /*y*/ /*y*/'
```

Does that make sense? Do you see an other way around this issue?